open_prices/api/proofs/views.py

```diff
@@ -140,12 +143,20 @@ def upload(self, request: Request) -> Response:
         # validate
         serializer = ProofCreateSerializer(data=proof_create_data)
         serializer.is_valid(raise_exception=True)
-        # get owner & source
-        owner = (
-            self.request.user.user_id
-            if self.request.user.is_authenticated
-            else settings.ANONYMOUS_USER_ID
-        )
+
+        # get owner (we allow anonymous upload, only if token is not present)
+        if self.request.user.is_authenticated:
+            owner = self.request.user.user_id
+        else:
+            if has_token_from_cookie_or_header(self.request):
+                return Response(
+                    {
+                        "detail": "Authentication failed. Please pass a valid token, or remove it to upload the proof anonymously."
+                    },
+                    status=status.HTTP_400_BAD_REQUEST,
```

Comment:
I opted for 400 instead of 403, because the endpoint doesn't *always* require authentication ^^